mousestyles/data/__init__.py

```diff
@@ -62,3 +62,105 @@ def load_all_features():
     other_features = [x.iloc[:, -1] for x in data_frames]
     other_features.insert(0, all_features_df)
     return pd.concat(other_features, axis=1)
+
+
+def load_intervals(feature):
+    """
+    Parameters
+    ----------
+    feature: {"AS", "F", "IS", "M_AS", "M_IS", "W"}
+
+    Returns
+    -------
+    intervals : pandas.DataFrame
+        All the data from the feature directory into one data frame
+        columns are strain, mouse, day, start, stop
+
+    Examples
+    --------
+    >>> AS = load_intervals('AS')
+    >>> IS = load_intervals('IS')
+    """
+    # check input is one of the provided choices
+    if feature not in {"AS", "F", "IS", "M_AS", "M_IS", "W"}:
+        raise ValueError(
+            'Input value must be one of {"AS", "F", "IS", "M_AS", "M_IS", "W"}'
+            )
+    # get all file names
+    file_names = _os.listdir(_os.path.join(data_dir, "intervals", feature))
+    # check if directory is empty
+    if len(file_names) == 0:
+        raise ValueError('Directory is empty; no file found.')
+    # initialized data frame
+    # use for loop to load every file and concat to overall data frame
+    dt = pd.DataFrame()
+    for item in file_names:
+        strain = int(item.split("strain")[1].split("_mouse")[0])
+        mouse = int(item.split("mouse")[1].split("_day")[0])
+        day = int(item.split("day")[1].split(".npy")[0])
+        path = _os.path.join(data_dir, "intervals", feature, item)
+        sub = np.load(path)
+        dt_sub = pd.DataFrame()
+        dt_sub["strain"] = [strain] * sub.shape[0]
+        dt_sub["mouse"] = [mouse] * sub.shape[0]
+        dt_sub["day"] = [day] * sub.shape[0]
+        dt_sub["start"] = sub[:, 0]
+        dt_sub["stop"] = sub[:, 1]
+        dt = pd.concat([dt, dt_sub])
+    # sort based on strain, mouse and day
+    dt = dt.sort(["strain", "mouse", "day"])
```

Comment:
I think it looks better when the return data frame is sorted already, since some of the analysis is base on mouse-day. I timed it and it takes less than 1 second to sort a 4260849 row data frame.